### PR TITLE
Fix ProductPrice and ProductImage behavior in edge cases

### PR DESCRIPTION
**Pull request by estacioneto**
#### What is the purpose of this pull request?

- Do not render `ProductPrice` when selling price is zero.
- Remove content loader from `ProductImage` and put a placeholder when the product has no image.

#### What problem is this solving?

- Showing selling price even when it's zero.
- `ProductImage` was showing content loader but the product had no image.

#### How should this be manually tested?

[Access the workspace](https://summary--invictastores.myvtex.com/impact-cases)

#### Screenshots or example usage

image

#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/legacy/components/ProductSummarySmall.js [hunk 1]
```diff
@@ -32,12 +31,17 @@ class ProductSummarySmall extends Component {
       'overflow-hidden br3 h-100 flex flex-column justify-between center tc'
     )
 
-    const summaryClasses = classNames(`${productSummary.element} pointer pt3 pb4 flex flex-column`, {
-      'bb b--muted-4 mh2 mt2': showBorders,
-    })
+    const summaryClasses = classNames(
+      `${productSummary.element} pointer pt3 pb4 flex flex-column`,
```
Comment (lucasecdb) [inline react/legacy/components/ProductSummarySmall.js]: nit: no need for template string when using classNames

File: react/legacy/components/ProductSummarySmall.js [hunk 2]
```diff
@@ -32,12 +31,17 @@ class ProductSummarySmall extends Component {
...
+      {
+        'bb b--muted-4 mh2 mt2': showBorders,
+      }
+    )
 
     const nameClasses = {
-      containerClass: `flex items-start ${productSummary.nameContainer} justify-center pv5 t-mini pb2`,
+      containerClass: `flex items-start ${
+        productSummary.nameContainer
```
Comment (lucasecdb) [inline react/legacy/components/ProductSummarySmall.js]: same

File: react/legacy/components/ProductSummarySmall.js (hunk 1 above)
Comment (estacioneto) [reply react/legacy/components/ProductSummarySmall.js]: Wasn't me 😢

File: react/legacy/components/ProductSummarySmall.js (hunk 2 above)
Comment (estacioneto) [reply react/legacy/components/ProductSummarySmall.js]: same 😢

File: react/legacy/components/ProductSummarySmall.js (hunk 1 above)
Comment (lucasecdb) [reply react/legacy/components/ProductSummarySmall.js]: sorry m8

Comment (brunojdo) [reply react/legacy/components/ProductSummarySmall.js]: doit

File: react/components/ProductSummaryImage/ProductImage.js [hunk 3]
```diff
@@ -65,15 +97,17 @@ const ProductImageContent = ({
...
+  )
 
   return compose(
     withBadge(showBadge),
     withCollection(showCollections)
   )(img)
 }
 
-const ProductImage : FunctionComponent = (props) => {
+const ProductImage = props => {
```
Comment (arthurepc) [inline react/components/ProductSummaryImage/ProductImage.js]: Any particular reason for this component's type to be removed? I saw that you're typing ProductSummaryPrice

Comment (estacioneto) [reply react/components/ProductSummaryImage/ProductImage.js]: The reason is that this file is not typescript 😁

Comment (arthurepc) [reply react/components/ProductSummaryImage/ProductImage.js]: hahahaha sure

Comment (estacioneto) [reply react/components/ProductSummaryImage/ProductImage.js]: I'll remove all typed components. Someone did it and I don't know why it didn't break the whole store

Comment (arthurepc) [reply react/components/ProductSummaryImage/ProductImage.js]: nice, it makes sense

**Pull request merged**